### PR TITLE
Fix the flake8 workflow script

### PR DESCRIPTION
**Pull request by Ellen99**
Updated the workflow script to install flake8.
After pushing this commit, the workflow for flake8 should pass.

There were ~85 format errors reported by flake8 in the following python scripts:
./auto-run-nondex/valid_repo_extractor.py
./auto-update-dataset/python/repoArchivedCheck.py
./format_checker/common_checks.py
./format_checker/forked_pr_remover.py
./format_checker/main.py:21:80
./format_checker/pr_checker.py
./format_checker/utils.py
./repo-info/get_repo_info.py

### CODE REVIEW

File: repo-info/get_repo_info.py [hunk 1]
```diff
@@ -8,27 +8,39 @@
 tqdm.pandas()
 
 parser = argparse.ArgumentParser()
-parser.add_argument('-t', '--github_access_token', help='GitHub access token to overcome API rate limitations')
-parser.add_argument('-f', '--filepath', help='Filepath of .csv file containing repo data')
-parser.add_argument('-c', '--colname', help='Column name in CSV file pertaining to repo URL')
+parser.add_argument('-t',
+                    '--github_access_token',
+                    help='GitHub access token to overcome API rate limitations'
+                    )
```
[the remark this answers is not in the document]
Comment (Ellen99) [reply repo-info/get_repo_info.py:14]: That line was 80 chars with parentheses, which is too long for the flake8(it requires the line to be <79),
I didn't want to break the string into parts, so I moved the closing parentheses to a new line.
`Flake8` primarily checks for code style and formatting issues but does not enforce a specific coding style.

The coding style preferences depend on us and how we want our code to look; there is no right or wrong.
_Before my recent changes, there wasn't a consistent coding style across all our Python files. That's why I didn't pay too much attention to it._
____
I have searched and found a common coding style for python: `PEP 8`. 
We can use `black` in combination with `Flake8` to automatically format the code according to specific style guidelines.

`PEP 8` recommends that the opening parentheses should be on the same line as the function or method name and the closing parentheses should be on the same line as the last argument.

Let me know if you want me to combine `black` with `flake8`.

File: format_checker/common_checks.py [hunk 2]
```diff
@@ -12,9 +12,12 @@
 )
 
 
-# Contains regexes for columns that are commmon to pr-data, gr-data and tic-fic-data
+# Contains regexes for columns that are \
```
[the remark this answers is not in the document]
Comment (Ellen99) [reply format_checker/common_checks.py]: fixed

File: auto-update-dataset/python/repoArchivedCheck.py [hunk 3]
```diff
@@ -5,23 +5,27 @@
 
 # it takes about 2 minutes to check 200 projects
 def main():
-    # get unique urls from pr-data.csv, we can also use cmd: `git pull -r ; cut -f1 -d, pr-data.csv | uniq`
-    pr_data_url = r"https://raw.githubusercontent.com/TestingResearchIllinois/idoft/main/pr-data.csv"
+    # get unique urls from pr-data.csv, we can also use \
```
[the remark this answers is not in the document]
Comment (Ellen99) [reply auto-update-dataset/python/repoArchivedCheck.py]: fixed

File: .github/workflows/flake8-checker.yml [hunk 4]
```diff
@@ -23,6 +23,12 @@ jobs:
...
           pip install -r repo-info/requirements.txt
 
+      - name: Install Flake8
+        run: |
+          pip install flake8
+
       - name: Run Flake8
         run: |
           flake8
+
```
Comment (darko-marinov) [inline .github/workflows/flake8-checker.yml:33]: Please remove these unnecessary changes.

File: auto-run-nondex/valid_repo_extractor.py [hunk 5]
```diff
@@ -6,7 +6,10 @@
 AUTHOR_URL_POSTFIX = 'language=java&q=&sort=&type='
 AUTHOR_URL = ''
 REPO_URL_PREFIX = 'https://github.com'
-PR_DATA_CSV_FILE_URL = 'https://raw.githubusercontent.com/TestingResearchIllinois/idoft/main/pr-data.csv'
+PR_DATA_CSV_FILE_URL = (
```
Comment (darko-marinov) [inline auto-run-nondex/valid_repo_extractor.py:9]: (How) can `flake8` be configured to allow longer lines?

File: auto-run-nondex/valid_repo_extractor.py [hunk 6]
```diff
@@ -25,11 +28,17 @@ def extract_org_repos_with_pom_file():
     repos = soup.find_all('a', class_='d-inline-block', href=True)
     while len(repos) > 3:
         for repo in repos:
-            if repo.has_attr('data-hovercard-type') and repo.get('data-hovercard-type') == 'repository':
```
Comment (darko-marinov) [inline auto-run-nondex/valid_repo_extractor.py:28]: Why not do just `get` instead of `has_attr` and `get`?

File: auto-update-dataset/python/repoArchivedCheck.py [hunk 7]
```diff
@@ -5,23 +5,27 @@
...
-    # get unique urls from pr-data.csv, we can also use cmd: `git pull -r ; cut -f1 -d, pr-data.csv | uniq`
-    pr_data_url = r"https://raw.githubusercontent.com/TestingResearchIllinois/idoft/main/pr-data.csv"
+    # get unique urls from pr-data.csv, we can also use
+    # cmd: `git pull -r ; cut -f1 -d, pr-data.csv | uniq`
+    pr_data_url = r"https://raw.githubusercontent.com/" \
+              r"TestingResearchIllinois/idoft/main/pr-data.csv"
     urls = pd.read_csv(pr_data_url, usecols=["Project URL"])
     urls = list(set([i[0] for i in urls.values.tolist()]))
     print("load data from", pr_data_url)
-
```
Comment (darko-marinov) [inline auto-update-dataset/python/repoArchivedCheck.py:13]: Please no unnecessary blank-space changes. If some are needed, make a separate PR with only those.

**Review by darko-marinov — CHANGES_REQUESTED** [review]
I've only skimmed over the first few changes. Most of them are due to line length. Can you first configure `flake8` to allow longer lines, then address all issues that are not about line length, and finally check if we want to shorten lines.

File: auto-update-dataset/python/repoArchivedCheck.py (hunk 7 above)
Comment (Ellen99) [reply auto-update-dataset/python/repoArchivedCheck.py:13]: This was a warning by Flake8, do we want to ignore or acknowledge it? 

_blank line contains whitespace_ Flake8(W293)

Comment (darko-marinov) [reply auto-update-dataset/python/repoArchivedCheck.py:13]: Fine to address that, but can we have separate commits for these cases:
1. blank lines and whitespace
2. other non-long line issues
3. long-line issues

Can `flake8` be configured to run only one check or some checks? Dually, can it be configured to not run some checks?

Comment (Ellen99) [reply auto-update-dataset/python/repoArchivedCheck.py:13]: yes we can configure the `flake8` to run only specific checks or a subset of checks.
We can use --select and --ignore options in our yml workflow file.
ex. : `flake8 --ignore E501` - to ignore line too long check

File: repo-info/get_repo_info.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ellen99) [reply repo-info/get_repo_info.py:14]: @darko-marinov , @zzjas 

Lets decide what we want to do with the parentheses styling,
- Do we want to enforce PEP8 or some other style over the repository or we keep as it is now?
- Additionally the longest line in the repository was 220 characters(it was a regex). Should I set max to 220 and just keep all lines as they are? I can also ignore that check in the workflow if we do not need particular length max for the lines.

File: auto-update-dataset/python/repoArchivedCheck.py (hunk 7 above)
Comment (Ellen99) [reply auto-update-dataset/python/repoArchivedCheck.py:13]: 1. PR for blank lines and whitespace : #781 
2. PR for other non-long line issues : #782 
3. I will prepare a separate PR for long lines, after [discussing details](https://github.com/TestingResearchIllinois/idoft/pull/779/files#r1336459947)

Comment (darko-marinov) [reply auto-update-dataset/python/repoArchivedCheck.py:13]: I accepted the two PRs. Let's update this one with the final changes.

File: auto-run-nondex/valid_repo_extractor.py (hunk 6 above)
Comment (Ellen99) [reply auto-run-nondex/valid_repo_extractor.py:28]: I removed the redundant check

File: auto-run-nondex/valid_repo_extractor.py (hunk 5 above)
Comment (Ellen99) [reply auto-run-nondex/valid_repo_extractor.py]: I've skipped the long line checks for now, I can configure it and allow longer lines if needed

File: .github/workflows/flake8-checker.yml (hunk 4 above)
Comment (Ellen99) [reply .github/workflows/flake8-checker.yml]: moved to PR #781

File: auto-update-dataset/python/repoArchivedCheck.py (hunk 7 above)
Comment (Ellen99) [reply auto-update-dataset/python/repoArchivedCheck.py]: This PR is updated with the latest changes, I'm awaiting [your decision regarding the styling of parentheses](https://github.com/TestingResearchIllinois/idoft/pull/779#discussion_r1336459947).
